userspace/libsinsp/threadinfo.cpp

```diff
@@ -899,23 +899,14 @@ std::string sinsp_threadinfo::get_cwd()
 
 void sinsp_threadinfo::set_cwd(const char* cwd, uint32_t cwdlen)
 {
-	char tpath[SCAP_MAX_PATH_SIZE];
 	sinsp_threadinfo* tinfo = get_main_thread();
 
 	if(tinfo)
 	{
-		sinsp_utils::concatenate_paths(tpath,
-			SCAP_MAX_PATH_SIZE,
-			(char*)tinfo->m_cwd.c_str(),
-			(uint32_t)tinfo->m_cwd.size(),
-			cwd,
-			cwdlen);
+		tinfo->m_cwd = sinsp_utils::concatenate_paths(m_cwd, std::string_view(cwd, cwdlen)).c_str();
```

Comment:
Please, remove the `c_str()` call. At first glance it would appear like a memory error (taking a pointer to the buffer of a temporary string), but it's not as I found that `m_cwd` is also a string.

Reply:
Mmmh, strange.
`tinfo->m_cwd` is a string, so I'd be keen on thinking that assigning a string or the pointer to its underlying buffer should end with the same outcome. Just out of curiosity have you tried what happens with this re-arrangements?

```c++
// uses the returned string
tinfo->m_cwd.assign(sinsp_utils::concatenate_paths(m_cwd, std::string_view(cwd, cwdlen)));

// uses the returned string's buffer pointer
tinfo->m_cwd.assign(sinsp_utils::concatenate_paths(m_cwd, std::string_view(cwd, cwdlen)).c_str());
```

Again, the code you wrote is correct. It just appears as it's assigning a temporary buffer to a more permanent object as it's not immediately evident that `tinfo->m_cwd` is a string, not a `char*` and therefore copies the buffer. Feel free to use the implementation you prefer, though it'd be interesting to solve the mystery of the missing slash.

Reply:
Only the c_str() version works. I added a comment ...

Reply:
I have refactored that function to take a `std::string_view` instead of a `const char*` + length (which is exactly the same thing) to hopefully get rid of this potential weird behavior.